ocaml/xapi/xapi_mgmt_iface.ml

```diff
@@ -198,13 +189,13 @@ let on_dom0_networking_change ~__context =
         debug "Changing Host.address in database to: %s" ip;
         Db.Host.set_address ~__context ~self:localhost ~value:ip;
         debug "Refreshing console URIs";
-        update_getty ();
+        Helpers.update_getty ();
         Dbsync_master.refresh_console_urls ~__context
       end
     | None ->
       if Db.Host.get_address ~__context ~self:localhost <> "" then begin
         debug "Changing Host.address in database to: '' (host has no management IP address)";
-        update_getty ();
+        Helpers.update_getty ();
         Db.Host.set_address ~__context ~self:localhost ~value:""
```

Comment:
But it seems that at startup we do update the inventory (`Xapi_mgmt_iface.change -> Xapi_inventory.update ...`) before this `Helpers.update_getty` is called. But yeah, this call order is quite confusing, but I think it works because we don't get the IP from the Db when we update the logon message.